src/mono/netcore/System.Private.CoreLib/src/System/WeakReference.T.Mono.cs

```diff
@@ -29,7 +29,7 @@ private void Create(T target, bool trackResurrection)
         {
             if (trackResurrection)
             {
-                trackResurrection = true;
+                this.trackResurrection = true;
```

Comment:
This looked like a real bug.  Please let me know if something different should be done.

Reply:
Yes looks like a bug that goes back to https://github.com/mono/mono/commit/30a2eeb98275a1432dc990ef6beee354e8bb2842.

I checked that we don't have the issue in the classic mono/mono implementation.